source/expressions.tex

```diff
@@ -666,7 +666,7 @@
 the glvalue.
 
 \item Otherwise, if the object to which the glvalue refers contains an invalid
-pointer value\iref{basic.stc.dynamic.deallocation}, the behavior is
+pointer value\iref{basic.compound}, the behavior is
```

Comment:
What about here, shouldn't this also refer to basic.stc.general for the definition of "invalid pointer"? Why is basic.compound a good reference?

Reply:
But [basic.stc.general]/4 specifies how to form an invalid pointer value, while [basic.compound]/3.4 is the definition (so [basic.stc.general]/4 refers to [basic.compound]/3.4).

And I think

> Indirection through an invalid pointer value and passing an invalid pointer value to a deallocation function have undefined behavior. Any other use of an invalid pointer value has implementation-defined behavior.

in [basic.stc.general]/4 should be better placed in [basic.compound]/3.  But [basic.compound]/3 may need to be split because it is too long.

Reply:
Sorry, could you explain a bit how basic.general is the definition? I don't see "invalid" mentioned in basic.general except for in that list in p3 (which doesn't seem like a definition?) and in notes.

Reply:
[basic.compound] p3 contains the taxonomy of pointer values. It states "there exists a pointer value that is an invalid pointer value".

[basic.stc.general] p4 tells us about one way how to obtain an invalid pointer value (maybe there are others). It also explains a bit of the semantics of an invalid pointer value, but this should probably be moved elsewhere in the long run. For example, [conv.lval] p3.3 tells us a little more about an invalid pointer value.

Reply:
Yes, right. In fact, [conv.lval] is the very subject of this PR. I'm not sure where the definition of invald value could be moved to improve the presentation; isn't basic.stc.general a good place? We could just add a cross reference to it from [basic.compound]/(3.4).

Reply:
[basic.stc.general] shouldn't talk about indirection through an invalid pointer value; that should be handled by unary "operator*".

Reply:
Sure, but that's converted to use "operator*" (expr.ref p2 and expr.mptr.oper p3).

Reply:
> Maybe the last two sentences of [[basic.stc.general]p4](https://eel.is/c++draft/basic.stc.general#4) should become a note?

Except that we seem to lack normative statements elsewhere.

Reply:
Yes, the normative wording would have to move somewhere more appropriate.

Reply:
CWG2822 moves the normative wording "somewhere else", for unrelated reasons.

Reply:
Shall we wait for that to land then?